src/components/experimental/IndexProvider/utilities/types.ts

```diff
@@ -0,0 +1,31 @@
+export type IndexSelectedItems = string[] | 'All';
+
+export const SELECT_ALL_ITEMS = 'All';
+
+export enum SelectionType {
```

Comment:
I've been having some confusing feelings about exported enums recently when compared to using union types. 

I can't yet fully articulate it but I think it boils down to: "importing a new object instead of using a string value is slight extra complication. It adds surface area to our API and it is slightly harder to teach enums compared to 'onSelectionChange's first argument is a string that is either `all`, `page` `multi` or `single`". It's not a big difference (which is why I'm finding it hard to have a strong opinion) but I think I'm slightly leaning towards union types.

We tend to use union types in almost all other places (PopoverCloseSource, and DatePicker are the current exceptions, and [I'm looking at removing that in DatePicker](https://github.com/Shopify/polaris-react/pull/3113)) - e.g Button's size prop is `slim | medium | large` not `enum ButtonSize { Slim = "slim", Medium = "medium", Large = large}1


What do you think about using a union type `type SelectionType = 'all' | 'page' | 'multi' | single` instead?

- You can't iterate through all possible options at runtime (but we don't need to do this)
-